app/controllers/RouteController.scala

```diff
@@ -0,0 +1,163 @@
+package controllers
+
+import common.Serialization.{INSTANCE => Json}
+import common.{ActionFailed, Admin, BadRequest, Driver, NotFoundError}
+import dao.SessionDao
+import javax.inject.Inject
+import play.api.mvc.{AnyContent, MessagesControllerComponents, Request}
+import security.UserAction
+import services.RouteService
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class RouteController @Inject()(routeService: RouteService,
+                                sessionDao: SessionDao,
+                                cc: MessagesControllerComponents,
+                                override val userAction: UserAction
+                               )(implicit ec: ExecutionContext)
+  extends SecuredController(cc, userAction) {
+
+  def getCurrentRoute() = securedAsync(Driver.INSTANCE :: Nil, Action.async {
+    request: Request[AnyContent] => {
+      currentUser(request).flatMap {
+        case Some(user) =>
+          logger.info(s"returning current route for user ${user.getId}")
+          routeService.getCurrentRoute(user.getId).map {
+            case Some(route) =>
+              logger.info(s"returning route ${route.getId} with ${route.getPoints.size()} point to user ${user.getId}")
+              Ok(Json.toJson(route))
+            case None =>
+              logger.info(s"Could not find route for user ${user.getId}")
+              NotFound(Json.toJson(new NotFoundError("There's no route for this user")))
+          }
+        case None =>
+          logger.error(s"User session was verified, but now no user found, THIS SHOULD NEVER HAPPEN")
+          Future.successful(InternalServerError(Json.toJson(
+            new common.InternalServerError("No user found for existing session, this should never happen"))))
+      }
+    }
+  })
+
+  def getRoute(id: Long) = securedAsync(Driver.INSTANCE :: Admin.INSTANCE :: Nil, Action.async {
+    request: Request[AnyContent] => {
+      currentUser(request).flatMap {
+        case Some(user) =>
+          logger.info(s"returning route with id $id")
```

Comment:
`...for user ${user.getId}`?

Reply:
it's in later logs, it's ok to just say route here